### PR TITLE
E2E Module Setup Flow | Analytics: No existing account, no existing tag

### PR DESCRIPTION
**Pull request by circlecube**
## Summary

Addresses issue #356 

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by aaemnnosttv** [comment]
@googlebot I consent.

**Comment by aaemnnosttv** [comment]
Tested that this works as expected now with changes introduced in #154

File: tests/e2e/specs/modules/analytics/setup-no-account-no-tag.test.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		await expect( page ).toClick( '.mdc-tab', { text: /connect more services/i } );
+		await page.waitForSelector( '.googlesitekit-settings-connect-module--analytics' );
+
+		await Promise.all( [
+			page.waitForSelector( '.googlesitekit-setup-module__action .mdc-button' ),
+			expect( page ).toClick( '.googlesitekit-cta-link', { text: /set up analytics/i } ),
+		] );
+
+		// Intercept the call to window.open and call our API to simulate a created account.
+		await page.evaluate( () => {
```
Comment (swissspidy) [inline tests/e2e/specs/modules/analytics/setup-no-account-no-tag.test.js]: Is there any other way to do this without overriding `window.open`; perhaps by listening to `targetcreated`?

File: tests/e2e/plugins/module-setup-analytics-no-account.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Plugin Name: E2E Tests Module Setup Analytics API Mock No Account
+ * Plugin URI:  https://github.com/google/site-kit-wp
+ * Description: Utility plugin for mocking Analytics Setup API requests during E2E tests.
+ * Author:      Google
+ * Author URI:  https://opensource.google.com
+ */
```
Comment (swissspidy) [inline tests/e2e/plugins/module-setup-analytics-no-account.php]: Might wanna add the license info here, like #416 did for the other utility plugins already.

File: tests/e2e/specs/modules/analytics/setup-no-account-no-tag.test.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		await activatePlugin( 'e2e-tests-oauth-callback-plugin' );
+		await activatePlugin( 'e2e-tests-module-setup-analytics-api-mock-no-account' );
+	} );
+
+	afterEach( async() => {
+		await deactivateAllOtherPlugins();
+		await resetSiteKit();
+	} );
+
+	it( 'display create account when user has no analytics account', async() => {
```
Comment (swissspidy) [inline tests/e2e/specs/modules/analytics/setup-no-account-no-tag.test.js]: Minor nitpick, but think of `it` here as starting a sentence. Example:

`it ... displays account creation form when user has no Analytics account`

File: tests/e2e/specs/modules/analytics/setup-no-account-no-tag.test.js (hunk 1 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/modules/analytics/setup-no-account-no-tag.test.js]: I looked into it, but there is a known issue with intercepting requests in new tabs in Puppeteer.

See https://github.com/GoogleChrome/puppeteer/issues/386#issuecomment-343059315

> We currently don't have a way to attach to targets as soon as they are created, so there is no way to set up request interception for new tabs that won't miss some early requests. This is a known issue.

File: tests/e2e/plugins/module-setup-analytics-no-account.php (hunk 2 above)
Comment (aaemnnosttv) [reply tests/e2e/plugins/module-setup-analytics-no-account.php]: Added

File: tests/e2e/specs/modules/analytics/setup-no-account-no-tag.test.js (hunk 3 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/modules/analytics/setup-no-account-no-tag.test.js]: Updated

**Pull request merged**